api-server/src/server/utils/donation.js

```diff
@@ -314,3 +315,43 @@ export async function createStripeCardDonation(req, res, stripe) {
   await createAsyncUserDonation(user, donation);
   return res.status(200).json({ isDonating: true });
 }
+
+export async function handleStripeCardUpdateSession(req, app, stripe) {
+  const {
+    user: { id }
+  } = req;
+
+  const { Donation } = app.models;
+  log('Updating stripe card for user: ', ObjectId(id));
+
+  /* eslint-disable camelcase */
+  try {
+    // multiple donations support should be added
+    const donation = await Donation.findOne({
+      where: { userId: ObjectId(id) }
+    });
+
+    if (!donation) throw Error();
+
+    const { customerId, subscriptionId } = donation;
+
+    // updating customer payment method is handled by webhook handler
+    const session = await stripe.checkout.sessions.create({
```

Comment:
Sorry for the stupid question, but does this return the same `subscription` id as we send it? I just want to be 100% sure that updating a subscription doesn't change the id!

Reply:
So this just creates the checkout session and the returned object does not include a subscription id. In the cloud function, we are updating the user's default_payment_method and the subscription is not affected in any way.

Reply:
Great, thanks for clarifying.